code/modules/reagents/chemistry/reagents_holder.dm

```diff
@@ -248,6 +248,20 @@
 
 	handle_reactions()
 
+// This proc increases the temp of reagents by `temp_increase` amount, up until the target temperature, and no more
+/datum/reagents/proc/change_reagent_temp(temp_increase, target_temp)
+	if(chem_temp == target_temp || !temp_increase) // We don't need to do the mathy math
+		return
+	if(target_temp) // If we have a target temp, we only go until that temperature
+		var/difference = target_temp - chem_temp
+		var/temperature_change = clamp(difference, -temp_increase, temp_increase)
+		chem_temp += temperature_change
```

Comment:
On a second thought, this will actually behave weirdly with mismatching directions of `temp_increase` and `target_temp`; I believe we can use `directional_bounded_sum()` to fix that and also simplify the code
```suggestion
		chem_temp = directional_bounded_sum(chem_temp, temp_increase, target_temp, target_temp)
```